### PR TITLE
feat: fee market

### LINKED ISSUE #164: Show Fee Market overview in Apps.

### LINKED ISSUE #165: Support get inactive relayers
Which `inactive` means:
```
If the collateral of any registered relayer is lower than required collateral threshold (e.g. slashed),
the enrolment of this relayer will become inactive(will be removed from the ask list, and not able to put new ask).
```
https://docs.darwinia.network/bridge-knowledge-base/bridge-economics#detailed-steps-in-implementation

- [ ] Get inactive relayers

### PR DESCRIPTION
**Pull request by JayJay1024**

### CODE REVIEW

File: src/components/feemarket/Orders.tsx [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+          ) {
+            return false;
+          }
+        }
+
+        if (block) {
+          if (!(item.startBlock === block || item.confirmBlock === block)) {
+            return false;
+          }
+        }
```
Comment (sxlwar) [inline src/components/feemarket/Orders.tsx]: Merge conditions
Do not mix expressions and statements together

File: src/components/feemarket/Orders.tsx [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+      orderId: node.id.split('-')[1],
+      deliveryRelayer: node.deliveredRelayerId?.split('-')[1],
+      confirmationRelayer: node.confirmedRelayerId?.split('-')[1],
+      assignedRelayer: node.assignedRelayerId?.split('-')[1],
+      startBlock: node.createBlock,
+      confirmBlock: node.finishBlock,
+      createTime: node.createTime,
+      finishTime: node.finishTime,
+      sender: node.sender,
+      confirmedSlotIndex: node.confirmedSlotIndex,
```
Comment (sxlwar) [inline src/components/feemarket/Orders.tsx]: Why not use the spread operator?

File: src/components/feemarket/Overview.tsx [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+
+  useEffect(() => {
+    const { date, data } = feeHistoryData?.orderEntities?.nodes.reduce(
+      ({ date, data }, { fee, createTime }) => {
+        date.push(createTime.split('.')[0].replace(/-/g, '/'));
+        data.push(fromWei({ value: fee }, prettyNumber));
+
+        return { date, data };
+      },
+      { date: [], data: [] } as { date: string[]; data: string[] }
```
Comment (sxlwar) [inline src/components/feemarket/Overview.tsx]: date should obviously be in plural

File: src/components/feemarket/Overview.tsx [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+      xAxis: {
+        type: 'category',
+        data: totalOrders.date,
+      },
+      yAxis: {
+        type: 'value',
+      },
+      series: [
+        {
+          // eslint-disable-next-line no-magic-numbers
```
Comment (sxlwar) [inline src/components/feemarket/Overview.tsx]: unused comment

File: src/components/widget/fee-market/index.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  value?: SegmentedType;
+  className?: string;
+}) => {
+  return (
+    <div className={`inline-flex items-center justify-center space-x-1 ${className}`}>
+      <span
+        className={`cursor-pointer bg-gray-300 px-2 rounded-l-sm ${value === SegmentedType.ALL ? 'bg-gray-400' : ''}`}
+        onClick={() => onSelect(SegmentedType.ALL)}
+      >
+        All
```
Comment (sxlwar) [inline src/components/widget/fee-market/index.tsx]: Missing translations?

**Review by sxlwar — CHANGES_REQUESTED** [review]

File: src/components/feemarket/Orders.tsx (hunk 1 above)
Comment (JayJay1024) [reply src/components/feemarket/Orders.tsx]: What does `Do not mix expressions and statements together` mean here.
I think returning early can save you from calculating the rest

Comment (sxlwar) [reply src/components/feemarket/Orders.tsx]:
```js
if (duration && !(duration[0].isBefore(item.createTime) && (!item.finishTime || duration[1].isAfter(item.finishTime)))) {
    return false;
}
```

Even better, we name the condition (e.g: !item.finishTime || duration[1].isAfter(item.finishTime)) to express the intent of the code

Comment (JayJay1024) [reply src/components/feemarket/Orders.tsx]: done

File: src/components/feemarket/Orders.tsx (hunk 2 above)
Comment (JayJay1024) [reply src/components/feemarket/Orders.tsx]: done

File: src/components/feemarket/Overview.tsx (hunk 3 above)
Comment (JayJay1024) [reply src/components/feemarket/Overview.tsx]: done

File: src/components/feemarket/Overview.tsx (hunk 4 above)
Comment (JayJay1024) [reply src/components/feemarket/Overview.tsx]: resolved

File: src/components/widget/fee-market/index.tsx (hunk 5 above)
Comment (JayJay1024) [reply src/components/widget/fee-market/index.tsx]: done

**Review by sxlwar — APPROVED** [review]

**Comment by ZolaxZola** [comment]
image

- [ ] Happens when Switching Network on Relayer Page

**Comment by JayJay1024** [comment]
> * Happens when Switching Network on Relayer Page

#260

**Pull request merged**